liquidprompt

```diff
@@ -924,48 +924,81 @@ _lp_git_branch_color()
             fi
         fi
 
-        local ret
-        local shortstat # only to check for uncommitted changes
-        shortstat="$(LC_ALL=C \git diff --shortstat HEAD 2>/dev/null)"
+        local has_changes="$(LC_ALL=C \git diff --shortstat HEAD 2>/dev/null)"
 
-        if [[ -n "$shortstat" ]]; then
-            local u_stat # shorstat of *unstaged* changes
-            u_stat="$(LC_ALL=C \git diff --shortstat 2>/dev/null)"
-            u_stat=${u_stat/*changed, /} # removing "n file(s) changed"
+        local diff_details=""
+        if (( LP_ENABLE_GIT_DIFF )); then
+            if [[ -n "$has_changes" ]]; then
+                local u_stat # shorstat of *unstaged* changes
+                u_stat="$(LC_ALL=C \git diff --shortstat 2>/dev/null)"
+                u_stat=${u_stat/*changed, /} # removing "n file(s) changed"
 
-            local i_lines # inserted lines
-            if [[ "$u_stat" = *insertion* ]]; then
-                i_lines=${u_stat/ inser*}
-            else
-                i_lines=0
-            fi
+                local i_lines # inserted lines
+                if [[ "$u_stat" = *insertion* ]]; then
+                    i_lines=${u_stat/ inser*}
+                else
+                    i_lines=0
+                fi
 
-            local d_lines # deleted lines
-            if [[ "$u_stat" = *deletion* ]]; then
-                d_lines=${u_stat/*\(+\), }
-                d_lines=${d_lines/ del*/}
-            else
-                d_lines=0
+                local d_lines # deleted lines
+                if [[ "$u_stat" = *deletion* ]]; then
+                    d_lines=${u_stat/*\(+\), }
+                    d_lines=${d_lines/ del*/}
+                else
+                    d_lines=0
+                fi
+
+                diff_details+=" +$i_lines/-$d_lines"
             fi
+        fi
 
-            local has_lines
-            has_lines="+$i_lines/-$d_lines"
+        local status_details=""
+        if (( LP_ENABLE_GIT_STATUS )); then
+            local status_lines="$(LC_ALL=C \git status --porcelain 2>/dev/null)"
 
-            if [[ -n "$has_commit" ]]; then
-                # Changes to commit and commits to push
-                ret="${LP_COLOR_CHANGES}${branch}${NO_COL}(${LP_COLOR_DIFF}$has_lines${NO_COL},$has_commit)"
-            else
-                ret="${LP_COLOR_CHANGES}${branch}${NO_COL}(${LP_COLOR_DIFF}$has_lines${NO_COL})" # changes to commit
+            if [[ -n "$status_lines" ]]; then
+                local staged="$(echo "${status_lines}" | \grep '^[^? ]. .*' | wc -l)"
+                local unstaged="$(echo "${status_lines}" | \grep '^.[^? ] .*' | wc -l)"
+                local untracked="$(echo "${status_lines}" | \grep '^?? .*' | wc -l)"
+
+                [[ "$staged" -gt 0 ]] && status_details+=" $LP_MARK_STAGED$staged"
```

Comment:
I would like to see lines like this where a string is put together to have the vairables wrapped in {}, like so:
" ${LP_MARK_STAGED}${staged}"

This on lines 964, 965, 970, 974, and 999